legend-engine-application-query/src/main/java/org/finos/legend/engine/application/query/api/QueryStoreManager.java

```diff
@@ -229,7 +231,7 @@ public List<Query> getQueries(QuerySearchSpecification searchSpecification, Stri
         }
         if (searchSpecification.taggedValues != null && !searchSpecification.taggedValues.isEmpty())
         {
-            filters.add(Filters.or(
+            filters.add(Filters.and(
```

Comment:
I don't think this is correct, the default behavior of search is `or`ing the conditions, you want a flag to customize this behavior, basically, `and` the tagged values condition together, maybe something like `SearchSpecification.combineTaggedValuesCondition: Boolean[0..1]`? Please make sure you document this well in code and Swagger